File: dotnet/src/SemanticKernel.Abstractions/Search/TextSearchOptions.cs
// Copyright (c) Microsoft. All rights reserved.
using System.Diagnostics.CodeAnalysis;

namespace Microsoft.SemanticKernel.Search;

/// <summary>
/// Options which can be applied when using <see cref="ITextSearch"/>.
/// </summary>
[Experimental("SKEXP0001")]
public sealed class TextSearchOptions
{
    /// <summary>
    /// Default number of search results to return.
    /// </summary>
    public static readonly int DefaultCount = 5;

    /// <summary>
    /// Flag indicating the total count should be included in the results.
    /// </summary>
    /// <remarks>
    /// Default value is false.
    /// Not all text search implementations will support this option.
    /// </remarks>
    public bool IncludeTotalCount { get; set; } = false;

    /// <summary>
    /// The basic filter expression to apply to the search query.
    /// </summary>
    public BasicFilterOptions? BasicFilter { get; set; }

    /// <summary>
    /// Number of search results to return.
    /// </summary>
    public int Count { get; set; } = DefaultCount;

    /// <summary>
    /// The index of the first result to return.
    /// </summary>
    public int Offset { get; set; } = 0;
}

File: python/semantic_kernel/data/text_search_options.py
# Copyright (c) Microsoft. All rights reserved.


from typing import Annotated

from pydantic import Field

from semantic_kernel.data.const import DEFAULT_COUNT
from semantic_kernel.data.filters.text_search_filter import TextSearchFilter
from semantic_kernel.data.search_options_base import SearchOptions


class TextSearchOptions(SearchOptions):
    """Options for a text search."""

    query: str | None = None
    filter: TextSearchFilter | None = None
    count: Annotated[int, Field(gt=0)] = DEFAULT_COUNT
    offset: Annotated[int, Field(ge=0)] = 0
    include_total_count: bool = False
